calculator.js

```diff
@@ -0,0 +1,76 @@
+function calc(...args) {
+    if (args.length < 3 || args.length % 2 === 0) {
+      throw new Error("Invalid input type");
+    }
+  
+    for (let i = 0; i < args.length; i += 2) {
+      if (typeof args[i] !== "number" || isNaN(args[i])) {
+        throw new Error("Invalid input type");
+      }
+      
+      if (i + 1 < args.length && typeof args[i + 1] !== "string") {
+        throw new Error("Invalid operator");
+      }
+    }
+  
+    const resultStack = [];
+    const operatorStack = [];
+  
+    const precedence = (operator) => {
+      switch (operator) {
+        case '+':
```

Comment:
let's use constants we created here for all cases like: 
```suggestion
        case PLUS_OPERATOR:
```

Reply:
Done